lib/global_config.js

```diff
@@ -278,4 +278,29 @@ exports.getKeepAlive = function () {
   return keepAlive;
 };
 
+let credentialManager = createJsonFileCredentialManager();
+
+exports.setCustomCredentialManager = function (customCredentialManager) {
+  Errors.checkArgumentValid(Util.checkValidCustomCrednetialManager(customCredentialManager),
+  ErrorCodes.ERR_GLOBAL_CONFIGURE_INVALID_CUSTOM_CREDENTIAL_MANAGER);
+  
+  credentialManager =  new function () {  
+    this.read = async function (key) {  
+      return await customCredentialManager.read(key);  
+    };  
+    this.write = async function (key, credential) {
+      await customCredentialManager.write(key, credential);
+    };  
+    this.remove = async function (key) {
+      await customCredentialManager.remove(key);
+    };  
+  };  
+};
+
+exports.getCredentialManager = function () {  
+  return credentialManager;  
+};  
 
+function createJsonFileCredentialManager() {
```

Comment:
Now that I think about it, IMO this would be cleaner: 
```
function createDefaultFileCredentialManager() {
  return require('./authentication/SecureStorage/jsonCredentialManager');
} 
```

Reply:
Done.